nilearn/_utils/estimator_checks.py

```diff
@@ -1190,43 +1192,115 @@ def check_masker_smooth(estimator):
 
         assert_array_equal(smoothed_signal, signal)
 
-    assert isinstance(signal, np.ndarray)
-    assert signal.shape == (n_sample, estimator.n_elements_)
-
 
 def check_masker_inverse_transform(estimator):
     """Check output of inverse transform.
 
     For signal with 1 or more samples.
 
-    Check that the proper error is thrown,
-    if signal has the wrong shape.
+    For nifti maskers:
+        - 1D arrays -> 3D images
+        - 2D arrays -> 4D images
+
+    For surface maskers:
+        - 1D arrays -> 2D images (Should we return 1D?)
```

Comment:
Why not 1D here ?

Reply:
good catch
seems I need to do another pass on the docstrings

Reply:
OK using doc strings templates because I am copy pasting the same thing across too many methods